wordpress-mobile/WordPress-iOS
# Empty Stats: Add event tracking for Publicize / Blogging Reminders

Part of #17307

## Description

This PR adds event tracking for the following events:
 
```
// Stats - Empty Stats nudges

stats_publicize_nudge_shown
stats_publicize_nudge_tapped
stats_publicize_nudge_dismissed
stats_publicize_nudge_completed
stats_blogging_reminders_nudge_shown
stats_blogging_reminders_nudge_tapped
stats_blogging_reminders_nudge_dismissed
stats_blogging_reminders_nudge_completed

// Stats - Customize card

stats_customize_insights_shown
```

## To test

To review this PR, please do the following:
1. Make sure events fire correctly by following the trigger in the table below
2. Validate the events (PCYsg-ktp-p2)

> What does it mean for an event to be valid?
> 
> An event can be marked valid when you’ve checked that it:
> 
>     Fires when it’s supposed to.
>     Fires exactly once each time it’s supposed to fire.
>     Has a blogid set if the event takes place in a site context.  If it’s not in a site context (e.g. a landing page or the wpcom homepage), you can omit a blogid.
>     Has the event properties that you intend it to have, and that are being attached to the event by your code.
> 
> Who should review it?
> 
> Similarly to code reviews, someone on your team or working in the same area as you is probably best to review your event.


## Event table

Event | Trigger | blog_id?
-- | -- | --
stats_publicize_nudge_shown | Publicize card is shown upon visiting the Stats Insights screen | ✅ 
stats_publicize_nudge_tapped  | Tapping on the `Enable post sharing` button on the Publicize card | ✅ 
stats_publicize_nudge_dismissed | Tapping on the `Dismiss` button on the Publicize card | ✅ 
stats_publicize_nudge_completed | Successfully setting up at least one connection via Publicize | ✅ 
stats_blogging_reminders_nudge_shown | Blogging Reminders card is shown upon visiting the Stats Insights screen | ✅ 
stats_blogging_reminders_nudge_tapped | Tapping on the `Set up blogging reminders` button on the Blogging Reminders card | ✅ 
stats_blogging_reminders_nudge_dismissed | Tapping on the `Dismiss` button on the Blogging Reminders card | ✅ 
stats_blogging_reminders_nudge_completed | Successfully setting up Blogging Reminders via the Blogging Reminders flow | ✅ 
stats_customize_insights_shown | Customize Insights card is shown upon visiting the Stats Insights screen | ❌ 

## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested and made sure events fire properly with the correct triggers / properties

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
